djsonb/lookups.py

```diff
@@ -72,10 +74,27 @@ def sql(self):
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, True)
                     else:
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, False)
-                    rule_specs.append(sql_tuple)
+                    # add to the list of rules generated for this pattern (one per field)
+                    got_pattern = patterns.get(pattern) or []
+                    got_pattern.append(sql_tuple)
+                    patterns[pattern] = got_pattern
+
+        rule_strings = [' AND '.join([rule[0] for rule in rule_specs])]
```

Comment:
That was the previous behavior, and it is very confusing. Okay, I'll go with ditching the intermediate list and naming the rule/pattern portions separately.